File: Sources/VihChatBotSDK/Models/DeviceTokenModel.swift
import Foundation

/// Body for the session-registry endpoint (architecture §3.3).
/// `platform` defaults to "ios" — Android sends "android".
public struct DeviceTokenRequest: Codable {
    public let deviceId: String
    public let fcmToken: String
    public let hashcode: String?
    public let platform: String
    public let sdkVersion: String

    public init(
        deviceId: String,
        fcmToken: String,
        hashcode: String?,
        platform: String = "ios",
        sdkVersion: String
    ) {
        self.deviceId = deviceId
        self.fcmToken = fcmToken
        self.hashcode = hashcode
        self.platform = platform
        self.sdkVersion = sdkVersion
    }

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case fcmToken = "fcm_token"
        case hashcode, platform
        case sdkVersion = "sdk_version"
    }
}

public struct DeviceTokenResponse: Codable {
    public var status: Bool
    public var message: String?
}

File: vihChatBot/src/main/java/com/vihmessenger/vihchatbot/data/model/DeviceTokenModel.kt
package com.vihmessenger.vihchatbot.data.model

import com.google.gson.annotations.SerializedName

/**
 * Request body for the session-registry endpoint introduced in architecture §3.3.
 * The backend persists {device_id -> fcm_token} in ElastiCache Redis so the
 * dispatch Lambda can decide between MQTT push and FCM wake-up.
 *
 * `hashcode` and `platform` give the server enough context to scope the entry per
 * tenant and per OS without an extra round-trip.
 */
data class DeviceTokenRequest(
    @SerializedName("device_id") val deviceId: String,
    @SerializedName("fcm_token") val fcmToken: String,
    @SerializedName("hashcode") val hashcode: String?,
    @SerializedName("platform") val platform: String = "android",
    @SerializedName("sdk_version") val sdkVersion: String
)

data class DeviceTokenResponse(
    @SerializedName("status") val status: Boolean,
    @SerializedName("message") val message: String? = null
)
